src/Microsoft.Extensions.Configuration.AzureAppConfiguration/AzureAppConfigurationExtensions.cs

```diff
@@ -70,8 +85,20 @@ public static IServiceCollection AddAzureAppConfiguration(this IServiceCollectio
                 throw new ArgumentNullException(nameof(services));
             }
 
-            services.AddLogging();
-            services.AddSingleton<IConfigurationRefresherProvider, AzureAppConfigurationRefresherProvider>();
+            bool providerDisabled = false;
+
+            try
+            {
+                providerDisabled = bool.TryParse(Environment.GetEnvironmentVariable(ConditionalProviderConstants.DisableProviderEnvironmentVariable), out bool disabled) ? disabled : false;
```

Comment:
Can we avoid reading the environment variable repeatedly?

Reply:
From what I understand it's only called during startup and could be repeated once in `UseAzureAppConfiguration`, and the only idea I had was storing it in the constants class I read from to get the environment variable name, but that didn't seem like the correct way to use the class. An exception is thrown as soon as `GetService` is called if the provider is disabled.

Reply:
I hope the environment variable is only read once and the decision is only made once. This ensures a consistent state. Yes, we can save it as a static variable at the startup to avoid rereading the environment variable.

I hope we can avoid reading the environment variable in the middleware too... Should we update the `AddAzureAppConfiguration(this IServiceCollection services)` too? I'm wondering if we can do something special there so the middleware knows.

Reply:
Its possible to keep all the environment variable logic in a single place by adding a new dummy refresher provider that implements `IConfigurationRefresherProvider`.

We only need to read the environment var once in the static constructor of `AzureAppConfigurationExtensions` class and store it in a private variable. Then we can use this variable to decide whether to add `AzureAppConfigurationSource` or skip it in `AddAzureAppConfiguration(this IConfigurationBuilder configurationBuilder,Action<AzureAppConfigurationOptions> action,bool optional = false)` method.
Additionally, we can add `EmptyRefresherProvider` instead of `AzureAppConfigurationRefresherProvider` in `AddAzureAppConfiguration(this IServiceCollection services)` method when provider is disabled. 

Then we dont need to read the variable in any other file and can eliminate the new constants file as well.

Reply:
Like it! Just one thing... We can probably do a stack method in `AzureAppConfigurationSource` instead of a static constructor. Failure of a static constructor can cause weird results.

Reply:
I'm not aware of what we could do with the `IServiceCollection` object, but I can look into it. I think the static variable makes sense.

Reply:
> We can probably do a stack method in `AzureAppConfigurationSource` instead of a static constructor.

@zhenlan Did you mean static method inside `AzureAppConfigurationExtensions`?

Reply:
Kinda weird to put it there, as it's not an extension method.

Reply:
`AzureAppConfigurationSource` is not static and we dont need to add a new instance unless provider is enabled. That's why I thought it might be best to either use static constructor or static field initializer within `AzureAppConfigurationExtensions`.